Sources/Hypertext.swift

```diff
@@ -49,8 +49,9 @@ extension Array: Renderable {
 }
 
 open class tag: Renderable {
-    public var name: String? = nil
-    public var isSelfClosing: Bool = false
+    open var isSelfClosing: Bool { return false }
+    open var name: String { return String(describing: type(of: self)) }
```

Comment:
Oh, okay. I believe it works! I am just trying to understand how. Does this work by taking exactly the name of the subclass? 

So if I did...
```swift
public class picture : tag {}
```

Then I could do...
```swift
picture().name // "picture"
```
because Swift is reading the exact name of the subclass?

I'm asking because I'm considering whether tying `name` directly to the name of the type is something we want to do. If we did, for example, there would be no way to create a custom subclass that you name something else, like `h2Special`, that actually renders to an `<h2></h2>` because you can't set `name` to whatever is desired.